### PR TITLE
[CPU] Refactored pushInput

### PR DESCRIPTION
**Pull request by mandrono**

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp [hunk 1]
```diff
@@ -94,44 +91,44 @@ void MKLDNNPlugin::MKLDNNInferRequest::InferImpl() {
             InferenceEngine::TBlob<float> *in_f = nullptr;
             switch (input.second->getTensorDesc().getPrecision()) {
```
Comment (myshevts) [inline inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp:92]: To me, the logical evolution of the refactoring would be moving (or even removing!) this huge (and straighforward) switch as well

Comment (dmitry-gorokhov) [reply inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp:92]: I agree. @mandrono please think about the solution

File: inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp [hunk 2]
```diff
@@ -41,105 +41,81 @@ MKLDNNPlugin::MKLDNNInferRequest::~MKLDNNInferRequest() {
...
-                case InferenceEngine::Precision::U64:
-                    // U64 is unsupported by mkldnn, so here we convert the blob and send I32
-                    copyConvert<int32_t>(InferenceEngine::Precision::I32, input, convertedInputs);
-                    break;
-                default:
-                    THROW_IE_EXCEPTION << "Unsupported input precision " << input.second->getTensorDesc().getPrecision();
-            }
-        }
-    }
+    // need to retain converted blobs until infer finish
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp:116]: In fact I am not sure we really need. As far as I can see before your changes `convertedInputs` was destroyed up to `graph->Infer(m_curBatch);`

File: inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp [hunk 3]
```diff
@@ -41,105 +41,81 @@ MKLDNNPlugin::MKLDNNInferRequest::~MKLDNNInferRequest() {
...
+                break;
+            }
+            case InferenceEngine::Precision::U8:
+            case InferenceEngine::Precision::BOOL:
+            case InferenceEngine::Precision::I16: {
+                if (graph->hasMeanImageFor(input.first))
+                    inPrec = InferenceEngine::Precision::FP32;
+                break;
+            }
+            // this precisions are unsupported by mkldnn, so here we convert the blob and send I32
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp:92]: these

Comment (dmitry-gorokhov) [reply inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp:92]: Please add equal comments for other groups of precisions

File: inference-engine/src/mkldnn_plugin/nodes/common/cpu_convert.cpp [hunk 4]
```diff
@@ -59,6 +59,9 @@ void convertFrom(void *srcPtr, void *dstPtr, Precision dstPrc, const size_t size
 }
 
 void cpu_convert(void *srcPtr, void *dstPtr, Precision srcPrc, Precision dstPrc, const size_t size) {
+    if (srcPtr == nullptr || dstPtr == nullptr)
+        THROW_IE_EXCEPTION << "cpu_convert has nullable data pointer";
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/common/cpu_convert.cpp:63]: `nullable` -> `null`

File: inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp [hunk 5]
```diff
@@ -211,7 +187,7 @@ void MKLDNNPlugin::MKLDNNInferRequest::GetBlob(const char *name, InferenceEngine
         // in situation if we push output blobs as inputs for next network (in Hetero plugin)
         // it may be that output tensor desc will be different from real input tensor desc for next network
         // because the optimal descriptor was chosen (e.g. inPlace case for Split node)
-        auto currBlockDesc = InferenceEngine::BlockingDesc(desc.getDims(), desc.getBlockingDesc().getOrder());
+        auto currBlockDesc = InferenceEngine::BlockingDesc(desc.getBlockingDesc().getBlockDims(), desc.getBlockingDesc().getOrder());
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp:190]: It looks like not refactoring, but changing the logic. Could you please clarify why the change is legal?

Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp:190]: At this moment previous version and current version are similar.
When we create ```output node``` we don't consider real tensor descriptor and create "default" ```MKLDNNMemoryDesc``` based only on layout.
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_input_node.cpp#L73
And blocked dimensions are equals to tensor dimensions.
According to ```BlockingDesc``` constructor its arguments are ```block_dims``` and ```order```.
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/inference_engine/ie_layouts.cpp#L238
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/mkldnn_memory.cpp#L1021

**Review by dmitry-gorokhov — APPROVED** [review]
